lib/mumukit/login/provider/cas.rb

```diff
@@ -0,0 +1,42 @@
+class Mumukit::Login::Provider::Cas < Mumukit::Login::Provider::Base
+  def configure_omniauth!(omniauth)
+    omniauth.provider :cas,
+                      url: cas_config.url,
+                      host: cas_config.host,
+                      ca_path: '.',
+                      disable_ssl_verification: true # FIXME this is big security issue
+  end
+
+  private
+
+  def cas_config
+    Mumukit::Login.config.cas
+  end
+end
+
+# ---
+
+module OmniAuth
+  module Strategies
+    class CAS
+      def callback_phase
+        # Monkey-patching to support phpCAS implementation
+        # where the first time the 'ticket' param is not sent.
+        if !on_sso_path? && !request.params['ticket']
+          return request_phase
+        end
+
+        # Original method
+        if on_sso_path?
```

Comment:
I'd use the alias_method idiom on order to avoid duplicating the gem's code:

```ruby
alias_method :__callback_phase__, :calback_phase
def callback_phase
      #...your patch...
      __callback_phase__
end
```